python/ffsim/qiskit/sim.py

```diff
@@ -715,6 +726,29 @@ def _evolve_state_vector_spinful(
         )
         return states.StateVector(vec=vec, norb=norb, nelec=nelec)
 
+    if isinstance(op, PermutationGate):
+        perm = list(op.pattern)
+        qs = qubit_indices
+        inverse_perm = [0] * len(perm)
+        for out_idx, src_idx in enumerate(perm):
+            inverse_perm[src_idx] = out_idx
+        dests = [qs[inverse_perm[i]] for i in range(len(qs))]
+        for src, dest in zip(qs, dests):
+            if (src < norb) != (dest < norb):
+                raise ValueError(
+                    f"Gate of type '{op.__class__.__name__}' must be applied on "
+                    "orbitals of the same spin."
+                )
```

Comment:
```suggestion
        if any(
            (src < norb) != (dest < norb) for src, dest in zip(qubit_indices, dests)
        ):
            raise ValueError(
                f"Gate of type '{op.__class__.__name__}' must be applied on "
                "orbitals of the same spin."
            )
```
removes one level of indentation.

Also, do you mind adding a test that triggers this case?

Reply:
Added a  `test_permutation_gate_cross_spin_error` to exercise the PermutationGate spin sector guard.
The test prepares a spinful Hartree–Fock state with `norb=1`, `nelec=(1, 0)`, applies `PermutationGate([1, 0])` to swap the alpha/beta qubits, and asserts `final_state_vector` raises the expected “same spin” ValueError. This directly triggers the `(src < norb) != (dest < norb)` condition.
However, I have concern that I created this standlone test function, and wonder whether I have better way to merge this test with other test functions.